src/components/Chat/ChatQueueComponent.tsx

```diff
@@ -1,27 +1,80 @@
-import React, { useMemo } from 'react';
+import React, { useContext, useMemo } from 'react';
+import { IGetMessage, ISocketMessage, IStudent } from '../../interfaces';
+import { addNewChat } from '../../reducers';
+import { RouteComponentProps, withRouter } from 'react-router';
+import {
+  createGenerateRoomMessage,
+  createGetQueueMessage,
+} from '../../services/message-service';
+import { SocketContext } from '../../providers';
 
-interface IProps {
-  queueMembers: string[];
-  createRoomWith(studentID: string): void;
-}
+const ChatQueueComponent = (props: RouteComponentProps) => {
+  const { history } = props;
+  const { queue, setQueue, uniqueID, dispatchChats, socketSend } = useContext(
+    SocketContext,
+  );
+
+  const toggleQueueMessage = (): void => {
+    // TODO: Toggle queue
+    const getMessage: IGetMessage = createGetQueueMessage();
+    socketSend(getMessage);
+  };
+
+  const createNewChatRoom = (student: IStudent) => {
+    if (student) {
+      dispatchChats(addNewChat(student));
+      setQueue(queue.filter(studentInQueue => studentInQueue !== student));
+      const socketMessage: ISocketMessage = createGenerateRoomMessage(
+        uniqueID,
```

Comment:
uniqueID burde kanskje skrives om til volunteerUniqueID el. for å være helt tydelig på forskjellen mellom student og frivillig sin id?

Reply:
Student sin vil alltid være denominert med "student" først